### PR TITLE
updated src duplicate-char, main, palindrome, rotation

### PR DESCRIPTION
**Pull request by NitinSaxenait**
Please use this PR template
What does this change do?
feature added of complex number and student operations
Any additional information for the reviewer to start
NA

How should this be manually tested?
We need window OS in which Rust is installed. then execute this rust program

Which Trello task and sub-task does this change relate to?

Are there any changes pending?
No

Does any team have to be notified of changes in this feature?
No, only for the mentor.

Definition of Done:

Is there >80% unit test code coverage?
if not please mention the Code Coverage of code.

Does this PR add new dependencies? If so, please list out the same.
No

Will this feature require a new piece of infrastructure to be implemented?
Is there appropriate logging included?
yes
Does the project compile ok?
yes

Have Clippy violations been fixed?
yes

Have Code is properly formatted?
yes

### CODE REVIEW

**Review by Dheerajjnv — DISMISSED** [review]

**Review by Dheerajjnv — APPROVED** [review]
Use log.

**Review by sonukumar-a11 — APPROVED** [review]
Please Write comment in more expressive way.

File: Assignment4/src/duplicate_character_string.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// code to find the repeated character in a string
+//
+// #Arguements
+//
+// a string is taken as a parameter to verify whether it has duplicate char or not
```
Comment (ksr30) [inline Assignment4/src/duplicate_character_string.rs:5]: redo

File: Assignment4/src/rotation_string.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+// #Arguements
+//
+// input1_string is an input string which is going to reverse
+// input2_string is an another string which is going to match to reverse of input1_string
+//
+// #Return
+//
+// return is going to return a string whether rotation is true or not
+
+pub fn reverse_is_true(input1_string: &str, input2_string: &str) -> String {
```
Comment (ksr30) [inline Assignment4/src/rotation_string.rs:12]: redo

**Review by ksr30 — CHANGES_REQUESTED** [review]

File: Assignment4/src/duplicate_character_string.rs (hunk 1 above)
Comment (NitinSaxenait) [reply Assignment4/src/duplicate_character_string.rs]: Done

File: Assignment4/src/rotation_string.rs (hunk 2 above)
Comment (NitinSaxenait) [reply Assignment4/src/rotation_string.rs]: Done

**Pull request closed without merging**